settings/src/components/webauthn/webauthn.js

```diff
@@ -29,23 +31,27 @@ export default function WebAuthn() {
 	// and then replace userkeys with the value returned by the call
 	// probably also refreshRecord( userRecord );
 	const onRegisterSuccess = useCallback( () => {
-		userKeys.push( {
-			id: 1111,
+		const newKeys = userRecord.record[ '2fa_webauthn_keys' ].push( {
+			id: Math.random(),
```

Comment:
Probably a very low risk of collisions?

Reply:
Yeah, it's just a temporary placeholder to demo the functionality for Design. My next PR will replace it entirely with an API request that will create the credential in the database. At that point it'll have a unique ID enforced on the server side. 

I just used `Math.random()` here a hardcoded value results in duplicate key warnings from React, since the component gets rendered multiple times.

Reply:
Props/Context in React components should be considered read-only. Should always avoid directly modifying props. Or it could make the code harder to understand and maintain, and may lead to subtle bugs. Perhaps change to something like the one below?

```js
const newKeys = [...userRecord.record[ '2fa_webauthn_keys' ], {id: Math.random(), name: 'New Key'}];
```

Reply:
That's a good note about Context being read only 👍🏻  I tried that code, but it didn't result in the new item being added to the list.

I don't think we need to worry about it, though, because that section was just a stub from #95. It wasn't intended to  be used for more than a few days, and it'll be replaced entirely by #200.

Reply:
I see. Thanks for clarifying 👍